firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptMiddleware.swift

```diff
@@ -36,7 +40,8 @@ class MicrosurveyPromptMiddleware {
     private func initializeMicrosurvey(windowUUID: WindowUUID, model: MicrosurveyModel) {
         let newAction = MicrosurveyPromptMiddlewareAction(
             windowUUID: windowUUID,
-            actionType: MicrosurveyPromptMiddlewareActionType.initialize(model)
+            actionType: MicrosurveyPromptMiddlewareActionType.initialize,
+            payload: model
```

Comment:
In some ways this is sort of bringing us back to what we had previously, except that the payloads were previously concrete subclasses of `ActionContext` and passed in as associated values. But we had used a similar idea of leveraging the benefits of inheritance in the custom payloads rather than having all of the properties declared on the (new) action classes.